compile.ts

```diff
@@ -163,8 +186,21 @@ class Compiler {
     } else if (ts.isIfStatement(n)) {
       this.compileIf(n);
     } else if (ts.isReturnStatement(n)) {
+      let values: ts.Expression[] = [];
       if (n.expression) {
-        this.#error("unsupported return expression", n.expression);
+        if (ts.isArrayLiteralExpression(n.expression)) {
+          n.expression.elements.map((el) => {
+            values.push(el);
+          });
+        } else {
+          values.push(n.expression);
+        }
+        if (values.length > this.outputs.length) {
```

Comment:
What I mean is that this could just initialize another output instead of returning an error. That way you can write a function without declaring the outputs:
```js
function foo() {
  return 1
}
```

Reply:
I see. This is not completely trivial, as I need to insert the input back at the right place in the instruction sequence and this could introduce some issue with references. I'll see how easy it is to do.

Reply:
Ok, didn't realize one you add a parameter at any place in the asm. Done.